board/safety/safety_subaru.h

```diff
@@ -1,18 +1,32 @@
-#define SUBARU_STEERING_LIMITS_GENERATOR(steer_max, rate_up, rate_down)          \
-  {                                                                              \
-    .max_steer = (steer_max),                                                    \
-    .max_rt_delta = 940,                                                         \
-    .max_rt_interval = 250000,                                                   \
-    .max_rate_up = (rate_up),                                                    \
-    .max_rate_down = (rate_down),                                                \
-    .driver_torque_factor = 50,                                                  \
-    .driver_torque_allowance = 60,                                               \
-    .type = TorqueDriverLimited,                                                 \
-  }                                                                              \
-
-const SteeringLimits SUBARU_STEERING_LIMITS      = SUBARU_STEERING_LIMITS_GENERATOR(2047, 50, 70);
-const SteeringLimits SUBARU_GEN2_STEERING_LIMITS = SUBARU_STEERING_LIMITS_GENERATOR(1000, 40, 40);
+#define SUBARU_STEERING_LIMITS_GENERATOR(steer_max, rate_up, rate_down)             \
+    .max_steer = (steer_max),                                                       \
+    .max_rt_delta = 940,                                                            \
+    .max_rt_interval = 250000,                                                      \
+    .max_rate_up = (rate_up),                                                       \
+    .max_rate_down = (rate_down),                                                   \
+    .driver_torque_factor = 50,                                                     \
+    .driver_torque_allowance = 60,                                                  \
+    .type = TorqueDriverLimited,                                                    \
+
+
+// the EPS will temporary fault if the steering rate is too high, so we cut   
+// the  steering torque every 7 frames for 1 frame if the steering rate is high
+#define SUBARU_STEERING_RATE_LIMIT_GENERATOR()                                      \
+    .min_valid_request_frames = 7,                                                  \
+    .max_invalid_request_frames = 1,                                                \
+    .min_valid_request_rt_interval = 144000,                                        \
+    .has_steer_req_tolerance = true,                                                \
```

Comment:
can we add to the above? macro name is confusing

Reply:
done